orchagent/intfsorch.cpp

```diff
@@ -174,20 +199,44 @@ void IntfsOrch::doTask(Consumer &consumer)
             {
                 SWSS_LOG_ERROR("Failed to remove action trap route ip:%s %d\n", ip_prefix.getIp().to_string().c_str(), status);
                 it++;
+                continue;
+            }
+
+            SWSS_LOG_NOTICE("Remove packet action trap route ip:%s\n", ip_prefix.getIp().to_string().c_str());
+            decreaseRouterIntfsRefCount(alias);
+            m_syncdIntfses[alias].ip_addresses.remove(ip_prefix.getIp());
+            it = consumer.m_toSync.erase(it);
+        }
+    }
+    cleanUpRouterInterfaces();
```

Comment:
This needs to be moved into the DEL_COMMAND block and the task cannot be erased until the router interface is cleaned up.